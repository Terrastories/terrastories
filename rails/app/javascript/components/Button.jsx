import React from "react";
import PropTypes from 'prop-types';

const Button = props => {
  const { buttonText, handleClick, buttonType } = props;

  return(
    <button type="button" className={buttonType} onClick={handleClick}>
      {buttonText}
    </button>
  )
}

Button.propTypes = {
  buttonText: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  buttonType: PropTypes.string.isRequired
};

export default Button;