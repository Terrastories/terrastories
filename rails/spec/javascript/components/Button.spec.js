import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../app/javascript/components/Button';

describe('Button component', () => {
  let buttonType = 'explore-button'; 
  let buttonText = 'EXPLORE';
  let wrapper;

  beforeEach(() => {
    global.I18n = {
      t: jest.fn(),
    },
    wrapper = shallow(
      <Button 
        buttonType={buttonType} 
        handleClick={global.I18n.t} 
        buttonText={buttonText} 
      />
    );
  });

  it('Displays correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.explore-button')).toHaveLength(1);
    expect(wrapper.text()).toEqual('EXPLORE');
  });

  it('Clicks the button', () => {
    wrapper.find('button').simulate('click')

    expect(global.I18n.t).toHaveBeenCalled();
  })
})