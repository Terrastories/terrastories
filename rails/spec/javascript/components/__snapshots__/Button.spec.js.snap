// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Displays correctly 1`] = `
<button
  className="explore-button"
  onClick={[MockFunction]}
  type="button"
>
  EXPLORE
</button>
`;
